src/OrleansRuntime/Streams/PersistentStream/PersistentStreamPullingAgent.cs

```diff
@@ -577,6 +579,15 @@ private void StartInactiveCursors(StreamConsumerCollection streamData)
 
         private async Task<bool> ErrorProtocol(StreamConsumerData consumerData, Exception exceptionOccured, bool isDeliveryError, IBatchContainer batch, StreamSequenceToken token)
         {
+            // for loss of client, we just remove the subscription
+            if (exceptionOccured is ClientNotAvailableException)
+            {
+                logger.Warn((int)ErrorCode.Stream_ConsumerIsDead,
+                    "Consumer {0} on stream {1} is no longer active - permanently removing Consumer.", consumerData.StreamConsumer, consumerData.StreamId);
```

Comment:
You will thank me when you get this log from client or prod and will be scratching your head about what us going on. With this details it will jump right at you.